src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/AspNetCoreHostingDiagnosticListener.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using Microsoft.ApplicationInsights.AspNetCore.Extensions;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DiagnosticAdapter;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+namespace Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners
+{
+    internal class AspNetCoreHostingDiagnosticListener: IApplicationInsightDiagnosticListener
+    {
+        private readonly TelemetryClient _client;
```

Comment:
Fixed all issues and added rules file (btw. `develop` branch had 144 violations)